dist/init/linux-upstart/caddy.conf

```diff
@@ -1,18 +1,25 @@
-description "Caddy startup script"
+description "Caddy HTTP/2 web server"
 author "Mathias Beke"
 
 start on runlevel [2345]
 stop on runlevel [016]
 
+console logged
 
 setuid www-data
 setgid www-data
 
 respawn
 respawn limit 10 5
 
-limit nofile 4096 4096
+# Letsencrypt-issued certificates will be written to this directory.
+env HOME=/etc/ssl/caddy
+
+limit nofile 1048576 1048576
 
 script
-	exec /usr/bin/caddy -agree=true -conf=/etc/caddy/Caddyfile
-end script
\ No newline at end of file
+	mkdir -p /etc/caddy
+	mkdir -p /var/tmp/caddy
```

Comment:
If you didn't find a safe directory (is there anything in `/proc/self` we could use?) then use a random directory like this:

```bash
mktemp -d -t "caddy-run.XXXXXX"
``` 

Please note that `/var/tmp` has stronger persistency guarantees than `/tmp`, hence it is an even less ideal location!